frontend/src/components/Buttons/FilterButton.vue

```diff
@@ -1,324 +1,299 @@
-                    {{ type.label }}
-                  </v-list-item-title>
+              <v-list-item :value="type.name" :title="type.label">
+                <template #append="{ isActive }">
                   <v-list-item-action>
-                    <v-checkbox :model-value="active" />
+                    <v-checkbox-btn :model-value="isActive" />
                   </v-list-item-action>
                 </template>
               </v-list-item>
             </template>
-          </v-list-group>
-        </v-list>
+          </v-list>
+        </v-expansion-panel-text>
       </v-expansion-panel>
       <v-expansion-panel v-if="yearFilters.length > 0" :title="$t('years')">
-        <v-list dense class="filter-content">
-          <v-list-group
-            v-model="selectedYearFilters"
-            multiple
-            @change="emitFilterChange">
+        <v-expansion-panel-text>
+          <v-list
+            select-strategy="multiple"
+            class="filter-content"
+            v-model:selected="selectedYearFilters"
+            @update:selected="emitFilterChange">
             <template
               v-for="(year, yearIndex) in yearFilters"
               :key="`year-${yearIndex}`">
-              <v-list-item :value="year">
-                <template #default="{ active }">
-                  <v-list-item-title>
-                    {{ year }}
-                  </v-list-item-title>
-
+              <v-list-item :value="year" :title="year">
+                <template #append="{ isActive }">
                   <v-list-item-action>
-                    <v-checkbox :model-value="active" />
+                    <v-checkbox-btn :model-value="isActive" />
                   </v-list-item-action>
                 </template>
               </v-list-item>
             </template>
-          </v-list-group>
-        </v-list>
+          </v-list>
+        </v-expansion-panel-text>
       </v-expansion-panel>
     </v-expansion-panels>
   </v-menu>
 </template>
 
-<script lang="ts">
-import { defineComponent } from 'vue';
+<script setup lang="ts">
+import { ref, defineEmits, watch } from 'vue';
+import { useRoute } from 'vue-router';
 import { BaseItemDto, ItemFilter } from '@jellyfin/sdk/lib/generated-client';
 import { getFilterApi } from '@jellyfin/sdk/lib/utils/api/filter-api';
 import { sanitizeHtml } from '@/utils/html';
-import { useSnackbar } from '@/composables';
+import { useI18n } from 'vue-i18n';
+import { first } from 'lodash';
+import { useRemote, useSnackbar } from '@/composables';
 
-export default defineComponent({
-  props: {
-    collectionInfo: {
-      type: Object as () => BaseItemDto,
-      required: true
-    },
-    itemsType: {
-      type: String,
-      required: true
-    },
-    disabled: {
-      type: Boolean,
-      required: false,
-      default: false
-    }
+const route = useRoute();
+const remote = useRemote();
+const { t } = useI18n();
+const emit = defineEmits(['change']);
```

Comment:
Define emits properly. See how I did it here: https://github.com/jellyfin/jellyfin-vue/blob/vite/frontend/src/components/Layout/Carousel/Carousel.vue#L79

Reply:
I changed this. the `library/_viewid` page expects a `Record<string, string[]>` when receiving the event.
Is there a better way to do this? Can we change the expected type of object in the `library/_viewid` page? to different `string[]`s?